ractor/src/actor/mod.rs

```diff
@@ -146,12 +146,12 @@ pub trait Actor: Sized + Sync + Send + 'static {
     /// * `myself` - A handle to the [ActorCell] representing this actor
     /// * `state` - A mutable reference to the internal actor's state
     #[allow(unused_variables)]
-    async fn post_start(
+    fn post_start(
         &self,
         myself: ActorRef<Self::Msg>,
         state: &mut Self::State,
-    ) -> Result<(), ActorProcessingErr> {
-        Ok(())
+    ) -> impl Future<Output = Result<(), ActorProcessingErr>> + Send {
```

Comment:
Why not use `async fn` directly here? It will be desugared to `impl Future` automatically.

Reply:
Because it won't add the `Send` bound, which is required for multi-threaded runtimes. It's outlined in the Rust foundation blog post: https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#--impl-trait-in-public-traits